{
  Future<?> future=newFutureInstance();
  assertTrue(future.cancel(true));
  assertTrue(future.isCancelled());
  assertTrue(future.isDone());
  try {
    future.get();
    fail("Expected CancellationException");
  }
 catch (  CancellationException e) {
    assertNotNull(e.getCause());
  }
}
